BQ25895M.device.lib.nut

```diff
@@ -291,11 +241,88 @@ class BQ25895M {
     
     //-------------------- PRIVATE METHODS --------------------//
     
+    // Set the enable charging bit, charging will happen automatically
+    function _enableCharging() {
+        
+        local rd = _getReg(BQ25895M_REG03);
+        rd = rd | (1 << 4); // set CHG_CONFIG bit
+        
+        _setReg(BQ25895M_REG03, rd);
+        
+    } 
+    
+    // Set target battery voltage
+    function _setChargeVoltage(vreg) { 
+        
+        // Convert to mV
+        vreg *= 1000;
+        
+        // Check that input is within accepted range
+        if (vreg < 3840) {
+            // minimum charge voltage from device datasheet
+            vreg = 3840; 
+        } else if (vreg > 4608) {
+            // maximum charge voltage from device datasheet
+            vreg = 4608; 
+        }
+        
+        local rd = _getReg(BQ25895M_REG06);
+        rd = rd & ~(0xFC); // clear current limit bits
+        rd = rd | (0xFC & (((vreg - 3840) / 16).tointeger()) << 2); // 3840mV is the default offset, 16mV is the resolution
+
+        _setReg(BQ25895M_REG06, rd);
+
+        // Note: Register default is 4.352V, need to 
+        // start watchdog to keep any other setting
+        _kickWatchdog();
+        
+    }
+    
+    // Set fast charge current
+    function _setChargeCurrent(ichg) {
+        
+        // Check that input is within accepted range
+        if (ichg < 0) { // charge current must be greater than 0
+            ichg = 0;
+        } else if (ichg > 5056) { // max charge current from device datasheet
+            sichg = 5056;
+        }
+        
+        local rd = _getReg(BQ25895M_REG04);
+        rd = rd & ~(0x7F); // clear current limit bits
+        rd = rd | (0x7F & ichg / 64); // 64mA is the resolution
+    
+        _setReg(BQ25895M_REG04, rd);
+        
+        // Note: Register default is 2048mA, need to 
+        // start watchdog to keep any other setting
+        _kickWatchdog();
+        
+    }
+    
+    function _setChargeCurrentOptimizer() {
```

Comment:
_setChargeCurrentOptimizer should be updated to take a boolean "true" or "false", so this setting can be enabled or disabled.